server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -106,13 +111,16 @@
         super(name);
         final List<SearchShardIterator> toSkipIterators = new ArrayList<>();
         final List<SearchShardIterator> iterators = new ArrayList<>();
+        Map<String, OriginalIndices> originalIndices = new HashMap<>();
         for (final SearchShardIterator iterator : shardsIts) {
+            originalIndices.putIfAbsent(iterator.getClusterAlias(), iterator.getOriginalIndices());
```

Comment:
can you add an assertion that originalindices are equal for the same cluster alias?